Iris-Pipeline/steps/process.py

```diff
@@ -0,0 +1,93 @@
+"""
+Process step that is used for Data processing in the program
+"""
+from pathlib import Path
+from sklearn.preprocessing import LabelEncoder
+from sklearn.model_selection import train_test_split
+import mlflow
+import click
+import pandas as pd
+import yaml
+
+def label_encoding_method(
+    dataset_df: pd.DataFrame,
+    label_name: str
+    ) -> pd.DataFrame:
+    """
+    Encode the label
+    
+    Takes the DataFrame from pandas, along with the label_name from the config file,
+    then it uses those 2 parameters to encode the label column,
+    into more usable numerical labels
+    
+    Args:
+        dataset_df (pandas.DataFrame): The dataset that was read through pandas,
+            and assigned as a dataframe.
+        label_name (str): The label column name derived from the configuration file
+        
+    Returns:
+        pandas.Dataframe: Holds the data from Irisdata_raw.csv but,
+            with an encoded label column
+    """
+    label_encoder = LabelEncoder
+    dataset_df[label_name] = label_encoder.fit_transform(label_encoder,dataset_df[label_name])
```

Comment:
Just a warning.
Be mindful when you do these kind of operations, as you are modifying the original dataframe, not a copy of it.

Reply:
Understood, i have changed the methodology to make a copy of the dataset and encode that copy, in case the original dataset is needed